ctc_lib/CLI_interface.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ctc_lib
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+type CLIInterface interface {
+	PrintO(c *cobra.Command, args []string)
+	SetRun(func(c *cobra.Command, args []string))
+	GetCommand() *cobra.Command
+	ValidateCommand() error
+	IsRunODefined() bool
+	Init()
+}
+
+func Execute(ctb CLIInterface) (err error) {
+	defer errRecover(&err)
+
+	if err := ctb.ValidateCommand(); err != nil {
+		CommandExit(err)
+	}
+	ctb.Init()
+	if ctb.IsRunODefined() {
+		cobraRun := func(c *cobra.Command, args []string) {
+			ctb.PrintO(c, args)
+		}
+		ctb.SetRun(cobraRun)
+	}
+
+	err = ctb.GetCommand().Execute()
+
+	//Add empty line as template.Execute does not print an empty line
+	ctb.GetCommand().Println()
+	CommandExit(err)
```

Comment:
It's a little weird to collect the error, then call exit, then return it. Can CommandExit return the error of it doesn't exit for some reason? Then we can return CommandExit

Reply:
yeah, i think i wanted to define ExecuteE which returns Error and Execute() which does os.Exit.
  Forgot to make those changes.

Reply:
done